File: hw2/java/Meal.java
public enum Meal {
    Paella(1, 13.25),
    Wu_Hsiang_Chi(2, 10.00),
    Bogrács_Gulyás(3, 11.25),
    Spanokopita(4, 6.50),
    Moui_Nagden(5, 12.95),
    Sambal_Goreng_Udang(6, 14.95);

    private final int id;
    private final double price;

    private Meal(int id, double price) {
        this.id = id;
        this.price = price;
    }

    public int id() {
        return id;
    }

    public double price() {
        return price;
    }

    public static Meal getRandomMeal() {
        Meal[] meals = Meal.values();
        int randomIndex = (int) (Math.random() * meals.length);
        return meals[randomIndex];
    }
}

File: hw2/go/meal.go
package main

import "math/rand"

type Meal struct {
	id    int
	name  string
	price float64
}

var Meals = []Meal{
	{1, "Paella", 13.25},
	{2, "Wu_Hsiang_Chi", 10.00},
	{3, "Bogrács_Gulyás", 11.25},
	{4, "Spanokopita", 6.50},
	{5, "Moui_Nagden", 12.95},
	{6, "Sambal_Goreng_Udang", 14.95},
}

func (m *Meal) getPrice() float64 {
	return m.price
}

func (m *Meal) getName() string {
	return m.name
}

func (m *Meal) ID() int {
	return m.id
}

func getRandomMeal() Meal {
	return Meals[rand.Intn(len(Meals))]
}
